apps/docusaurus/docs/guides/randomness.md

```diff
@@ -0,0 +1,314 @@
+---
+title: "Aptos Roll: on-chain public instant randomness API"
+---
+
+## What does it do: a quick example
+
+### How randomness may have been obtained insecurely/overcomplicated
+
+Building a lottery system and pick a random winner from n participants is trivial, at least in the centralized world with a trusted server: the backend simply calls a `random.randint(0, n-1)` (this is a python example).
+
+Unfortunately, without an equivalent of `random.randomint()` in Aptos Move, building a dApp version of it was another story.
+
+One may have written a contract where the random numbers are sampled insecurely (e.g., from the blockchain timestamp):
+
+```rust
+module module_owner::lottery {
+    // ...
+
+    struct LotteryState {
+        players: vector<address>,
+        winner_idx: std::option::Option<u64>,
+    }
+
+    fun load_lottery_state_mut(): &mut LotteryState {
+        // ...
+    }
+
+    entry fun decide_winner() {
+        let lottery_state = load_lottery_state_mut();
+        let n = std::vector::length(&lottery_state.players);
+        let winner_idx = aptos_framework::timestamp::now_microseconds() % n;
+        lottery_state.winner_idx = std::option::some(winner_idx);
+    }
+}
+```
+
+NOTE: it is insecure in multiple ways:
+
+- a malicious user may bias the result by picking the transaction submission time;
+- a malicious validator can bias the result easily by selecting which block the `decide_winner` transaction goes to.
+
+Other dApps may have chosen to use a external secure randomness source
+(e.g., [drand](https://drand.love/)), which is typically a complicated flow:
+
+1. The participants agree on using a future randomness seed promised by the randomness source to determine the winner.
+2. Once the randomness seed is revealed, the clients fetch the it and derive the winner locally.
+3. One of the participants submits the seed and the winner on chain.
+
+```rust
```

Comment:
I would link to the drand example in `aptos-move/move-examples` here, and copy & paste some of the code from it.

Reply:
In Nextra, you can do code snippets directly imported, without copy pasting.  Let's try moving there

Reply:
separate PR?